guardian/prosemirror-elements
# Add uuids to nodes representing fields

## What does this change?

Adds uuids to nodes representing fields, for quick `update` checks within FieldViews.

Because the node we receive in `update` is [not necessarily the _right_ node](https://prosemirror.net/docs/ref/#view.NodeView.update), we must check to make sure that it is before continuing with updating our fieldView.

We are no longer able to rely on `sameMarkup` to run this check as a result of #43 – but this is a small optimisation regardless, as `sameMarkup` will run on every editor update, and comparing uuids is more efficient.

## How to test

To the user, this is a no-op – everything should work as expected. I've added automated tests to ensure that uuids are present on every node representing a fieldView, whether it was created programmatically (via `.create`) or via serialisation (via `parseDOM`).
